internal/manifests/manifestutils/params.go

```diff
@@ -17,7 +17,15 @@ type Params struct {
 
 // StorageParams holds storage configuration.
 type StorageParams struct {
-	S3 *S3
+	S3           *S3
+	AzureStorage *AzureStorage
+}
+
+// AzureStorage for Azure Storage.
+type AzureStorage struct {
```

Comment:
Would it make sense to consolidate all Azure storage params in a single struct?
* `manifestutils.AzureStorage`
* `manifests.AzureStorage`
* `config.azureStorage` (options.go)
* `config.AzureStorage` (configmap.go)

Reply:
I had a similar feeling but followed the same approach as for S3 in this project and as Loki Operator for all the storages.

Reply:
That's true. I'm not sure why in Tempo the approach was different.

Since is a refactor that should be done for S3 too... what do you think if I create a follow up task to address it there?

Reply:
Yep, follow-up task sounds good, thanks!